### ISSUE TITLE
Cannot generate from more than 1 graphql file: Fragment "x" is never used.

### ISSUE THREAD

**Issue by fluxxu** [dotansimha/graphql-code-generator#3807]
**Describe the bug**
We wanted to use multiple *.graphql files to split the operations code, but we found that as soon as anyone file defined a fragment that was not used by the other files, the generator would make an error, even if that fragment was used in the file that defined it:
```
Found 1 error

  ✖ generated/graphql.ts
    AggregateError: 
        GraphQLDocumentError: Fragment "PostFragment" is never used.
            at /Users/fluxxu/Projects/graphql-codegen-issue/documents/file2.graphql:7:1
        at Object.checkValidationErrors (/Users/fluxxu/Projects/graphql-codegen-issue/node_
modules/@graphql-toolkit/common/index.cjs.js:431:15)
```
It looks like the generator inserted the fragment automatically into all the files, but graphql-toolkit thinks the unused fragment is error.

**To Reproduce**
```
git clone https://github.com/fluxxu/graphql-codegen-issue
cd graphql-codegen-issue
yarn
yarn gen
```

1. My GraphQL schema:

```graphql
type Author {
  id: Int!
  firstName: String!
  lastName: String!
  posts(findTitle: String): [Post]
}

type Post {
  id: Int!
  title: String!
  author: Author!
}

type Query {
  posts: [Post]
}

schema {
  query: Query
}

```

2. My GraphQL operations:

`file1.graphql`
```graphql
query QueryPosts {
  posts {
    ...PostFragment
  }
}

fragment PostFragment on Post {
  id
}
```

`file2.graphql`
```graphql
query QueryPosts2 {
  posts {
    id
  }
}
```

3. My `codegen.yml` config file:

```yml
overwrite: true
schema: "schema.graphql"
documents: "documents/**/*.graphql"
generates:
  generated/graphql.ts:
    plugins:
      - "typescript"
```

**Expected behavior**
The file got generated without error.

**Environment:**

- OS: macOS 10.15.3 (19D76)
- `@graphql-codegen/...`:
```
├─ @graphql-codegen/cli@1.13.1
├─ @graphql-codegen/core@1.13.1
├─ @graphql-codegen/introspection@1.13.1
├─ @graphql-codegen/plugin-helpers@1.13.1
├─ @graphql-codegen/typescript@1.13.1
└─ @graphql-codegen/visitor-plugin-common@1.13.1
``` 
- NodeJS: v12.13.0

**Comment by riencoertjens** [comment]
I just got this same error when upgrading [graphql](https://www.npmjs.com/package/graphql) to `v15.0.0`

**Comment by tylerbarker** [comment]
Yep having the same issue. Can confirm no issues when using `14.5.8`

**Comment by kamilkisiela** [comment]
ardatan/graphql-toolkit#541

**Comment by dotansimha** [comment]
As @kamilkisiela said, this was fixed in `graphql-toolkit` and already released. It's waiting for a codegen release (in a few hours).

**Comment by dotansimha** [comment]
If someone needs it, you can use `1.13.2-alpha-83993e60.114` of codegen, it should work.

**Comment by dotansimha** [comment]
Fixed in v1.13.2
